### PR TITLE
Filtering - Add Detailed view of a saved filter

### PR DESCRIPTION
**Pull request by martinkimmings**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Views/ManageFilters/DeleteFilter.cshtml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿@using NHSD.GPIT.BuyingCatalogue.UI.Components.TagHelpers
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Views/ManageFilters/DeleteFilter.cshtml]: Let's remove this page for now as the story is currently being worked on

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/ManageFiltersController.cs [hunk 2]
```diff
@@ -157,6 +163,16 @@ public IActionResult CannotSaveFilter(string backLink)
             return View(model);
         }
 
+        [HttpGet("delete")]
+        public IActionResult DeleteFilter(int filterId)
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/ManageFiltersController.cs]: Let's remove this one for now also, same reason as below.

File: src/NHSD.GPIT.BuyingCatalogue.Services/Solutions/ManageFiltersService.cs [hunk 3]
```diff
@@ -76,6 +76,19 @@ public async Task<List<Filter>> GetFilters(int organisationId)
                 .ToListAsync();
         }
 
+        public Task<Filter> GetFilter(int filterId)
+        {
+            return dbContext.Filters.Where(o => o.Id == filterId)
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.Services/Solutions/ManageFiltersService.cs]: There's a risk of this resulting in a cartesian explosion - can be resolved by adding `AsSplitQuery()`.

Additionally, the `Where` can be dropped and the query can be added to the `FirstOrDefaultAsync`. Also worth adding `AsNoTracking`

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Models/ManageFilters/FilterDetailsModel.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                x => epics.Where(c => c.Capability.Id == x.Id).OrderBy(c => c.Name));
+        }
+
+        public string OrganisationName { get; set; }
+
+        public int FilterId { get; set; }
+
+        public Filter Filter { get; init; }
+
+        public Dictionary<string, IOrderedEnumerable<Epic>> GroupedCapabilities { get; init; }
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Models/ManageFilters/FilterDetailsModel.cs]: `IOrderedEnumerable` is being enumerated multiple times in the View. `ToList` on the query, line 29, should fix this.

File: tests/NHSD.GPIT.BuyingCatalogue.Services.UnitTests/Solutions/ManageFiltersServiceTests.cs [hunk 5]
```diff
@@ -548,5 +548,38 @@ public static void Constructors_VerifyGuardClauses()
...
+
+            filter.Organisation = organisation;
+            filter.OrganisationId = organisation.Id;
+            filter.Framework = framework;
+            filter.FrameworkId = framework.Id;
+
+            await context.SaveChangesAsync();
+
+            context.Filters.Add(filter);
+            await context.SaveChangesAsync();
```
Comment (ghost) [inline tests/NHSD.GPIT.BuyingCatalogue.Services.UnitTests/Solutions/ManageFiltersServiceTests.cs:594]: Do we need to clear the `ChangeTracker` here to ensure it's not returning a cached copy?

**Review by bjss-colin — CHANGES_REQUESTED** [review]

**Review by bjss-colin — APPROVED** [review]

**Review by imranbashir01 — APPROVED** [review]

**Pull request merged**